src/api/cvc4cpp.cpp

```diff
@@ -1614,12 +1629,20 @@ size_t Term::getNumChildren() const
   {
     return d_node->getNumChildren() + 1;
   }
+  if(isCastedReal())
```

Comment:
Why do we need this here? Shouldn't `getNumChildren` have the correct value?

Reply:
We should discuss this approach.  Basically, we are using `(to_real n)` for integer constant `n` to represent the real constant `n`. This means that we need the API to treat `(to_real n)` as a constant, meaning that the Node and Term have a different opinion about the # children. In particular, `(to_real n)` has 1 child according to the internal API and 0 children according to the external API.